airflow/providers/ftp/hooks/ftp.py

```diff
@@ -77,8 +77,10 @@ def describe_directory(self, path: str) -> dict:
         :param path: full path to the remote directory
         """
         conn = self.get_conn()
+        current_path = conn.pwd()
         conn.cwd(path)
         files = dict(conn.mlsd())
+        conn.cwd(current_path)
```

Comment:
What about providing the path to `mlsd` method instead? (https://pythontic.com/ftplib/ftp/mlsd):
 ```suggestion
         files = dict(conn.mlsd(path=path))
```